src/main/scala/it/pagopa/interop/agreementmanagement/model/persistence/Adapters.scala

```diff
@@ -68,7 +69,8 @@ object Adapters {
         consumerDocuments = oldAgreement.consumerDocuments,
         createdAt = dateTimeSupplier.get(),
         updatedAt = None,
-        consumerNotes = oldAgreement.consumerNotes
+        consumerNotes = oldAgreement.consumerNotes,
+        document = oldAgreement.document // TODO new agreement must be passed?
```

Comment:
I guess it depends on:
- does the document contain information related to the EService descriptor?
- is it considered a different agreement?
- should it contain the date of the upgrade?

If any of these answer is "yes", I think we should generate a new document

Reply:
It's a No because we need to freeze the contract at the moment of its first activation.
So maintaining the old document is the right choice.

Reply:
not sure if there are other questions that could help us, those are the only ones that came to my mind